hpx/runtime/threads/thread_init_data.hpp

```diff
@@ -81,6 +92,11 @@ namespace hpx { namespace threads
 #endif
 #if defined(HPX_HAVE_THREAD_PARENT_REFERENCE)
             parent_locality_id(0), parent_id(nullptr), parent_phase(0),
+#endif
+#ifdef HPX_HAVE_APEX
+        /* HPX_HAVE_APEX forces the HPX_HAVE_THREAD_DESCRIPTION
+         * and HPX_HAVE_THREAD_PARENT_REFERENCE settings to be on */
+            apex_data(apex_new_task(description,parent_id)),
```

Comment:
Shouldn't this rather (deep-)copy the data from the parameter desc?

Reply:
The description is deep-copied in the apex_new_task() call:
https://github.com/STEllAR-GROUP/hpx/blob/c2a3be669721168fbf59c70ef09fc229e285ffa9/src/util/apex.cpp#L22 through line 29